browser/views.py

```diff
@@ -681,27 +700,49 @@ def edit_investigator_record(request,person_id,research_id):
     person = get_object_or_404(Person, pk=person_id)
     research = get_object_or_404(Investigator, pk=research_id)
     template = "browser/investigator.html"
+    institution_id = research.institution.id if research.institution else None
 
     # Retreive investigator data already stored in database and display to the user
     form = InvestigatorForm(initial={'subject': research.subject, 'role': research.role,
-                                             'year': research.year})
+                                             'year': research.year, 'institution_search': research.institution,
+                                     'institution_id': institution_id})
+
+    context = {
+        'form': form,
+        'person': person,
+        'investigator_data': research
+    }
 
     if request.method == 'POST':
         form = InvestigatorForm(request.POST, instance=person)
+        context.update({
+            'form': form
+        })
         if form.is_valid():
+            if form.cleaned_data.get('institution_search'):
+                if not form.cleaned_data.get('institution_id'):
+                    # Check if the checkbox is checked
+                    if form.cleaned_data.get('create_institution'):
+                        institution = Institution.objects.create(
+                            name=form.cleaned_data.get('institution_search'),
+                            changed_by=request.user
+                        )
+                        form.cleaned_data["institution_id"] = institution.id
+                    else:
+                        messages.add_message(request, messages.ERROR,
+                                             'The above institute does not exist in the database. '
+                                             'Please check the checkbox below to create a new institute.')
+                        return render(request, template, context)
+                research.institution = Institution.objects.get(id=form.cleaned_data.get('institution_id'))
```

Comment:
why don't you do the same thing you do in the add method (use the institution you get in line 726)? also, duplicate code.